__tests__/implementation.test.js

```diff
@@ -292,7 +292,133 @@ test('batch endpoint (multiple requests)', async () => {
     });
 });
 
-test('batch endpoint (multiple requests, error handling)', async () => {
+test('batch endpoint that throws errors', async () => {
```

Comment:
++, nice test.

Could we add one more test when the number of items are not match in `request/response` then throw `BatchItemNotFound` Error

Reply:
We already have these tests:
- https://github.com/Yelp/dataloader-codegen/blob/6ce1089cb898bae22add2420052d6a1892b6bbbe/__tests__/implementation.test.js#L547
- https://github.com/Yelp/dataloader-codegen/blob/6ce1089cb898bae22add2420052d6a1892b6bbbe/__tests__/implementation.test.js#L605

I've gone ahead and asserted that these are actually `BatchItemNotFound` errors now tho :)

Reply:
we could probably do with some better organization of these tests :P